### PR TITLE
CLOUDSTACK-9799 - Unable to create volume from snapshots with "_" in solidfire

### PR DESCRIPTION
**Pull request by syed**
Steps to reproduce:

1. Create a volume using the SolidFire plugin
2. Take a snpshot of the volume
3. Create volume from snapshot with "_" in the name
4. Error

@mike-tutkowski This as well :)

### CODE REVIEW

**Comment by mike-tutkowski** [comment]
@syed Are you looking to get this into 4.10? If you are not, I have this corrected in a PR scheduled for 4.11. If you are, then I think you're missing a file (SolidFireUtil).

**Comment by syed** [comment]
@mike-tutkowski If you already have this in your PR, I can close this. Let me know!

**Comment by mike-tutkowski** [comment]
Yes, but the PR is scheduled for 4.11.

On Feb 23, 2017, at 10:58 AM, Syed Mushtaq Ahmed <notifications@github.com<mailto:notifications@github.com>> wrote:


@mike-tutkowski<https://github.com/mike-tutkowski> If you already have this in your PR, I can close this. Let me know!

File: engine/storage/datamotion/src/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java [hunk 1]
```diff
@@ -711,6 +711,7 @@ private void handleCreateVolumeFromSnapshotBothOnStorageSystem(SnapshotInfo snap
         }
         catch (Exception ex) {
             errMsg = ex.getMessage() != null ? ex.getMessage() : "Copy operation failed in 'StorageSystemDataMotionStrategy.handleCreateVolumeFromSnapshotBothOnStorageSystem'";
+            copyCmdAnswer = new CopyCmdAnswer(errMsg);
```
Comment (mike-tutkowski) [inline engine/storage/datamotion/src/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java]: @syed What I don't have in my version of this fix is this line of code. I can add it to mine.

Comment (syed) [inline engine/storage/datamotion/src/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java]: That would be great @mike-tutkowski. That makes sure that the snapshot goes back to a good state if the storage driver operation fails. I will close this request then.

**Pull request closed without merging**